grails-app/services/org/pih/warehouse/product/ProductDataService.groovy

```diff
@@ -0,0 +1,23 @@
+package org.pih.warehouse.product
+
+import grails.gorm.services.Query
+import grails.gorm.services.Service
+
+@Service(Product)
+interface ProductDataService {
+
+    void delete(String id)
+
+    Product save(Product product)
+
+    Product get(String id)
+
+    @Query("""Select p from Product p
+              left join fetch p.documents
+              left join fetch p.synonyms
+              left join fetch p.productCatalogItems pci
+              left join fetch pci.productCatalog
+              where p.id=$id""")
+    Product getWithDocumentsAndSynonymsAndProductCatalog(String id)
```

Comment:
for me it looks good, since it is a known convention, but I know there will be a discussion around that.
`getEagerly`/`getWithAssociations` or anything similar seems not appropiate to me, since we MIGHT have some other similar methods and this kind of name would be confusing, because as I mentioned somewhere else - such names would fit if we fetched ALL possible associations